File: articles/automated-reasoning-symbolic-ai-and-hybrid-systems/cpp/symbolic_rule_check.cpp
#include <iostream>
#include <set>
#include <string>
#include <vector>

int main() {
    std::set<std::string> facts = {"has_documentation", "logs_decisions"};
    std::vector<std::string> premises = {"has_documentation", "logs_decisions"};
    bool fires = true;
    for (const auto& premise : premises) {
        if (!facts.count(premise)) fires = false;
    }
    std::cout << "rule_fires=" << std::boolalpha << fires << "\n";
    return 0;
}

File: articles/automated-reasoning-symbolic-ai-and-hybrid-systems/go/symbolic_rule_check.go
package main

import "fmt"

func main() {
	facts := map[string]bool{"has_documentation": true, "logs_decisions": true}
	premises := []string{"has_documentation", "logs_decisions"}
	fires := true
	for _, premise := range premises {
		if !facts[premise] {
			fires = false
		}
	}
	fmt.Printf("rule_fires=%v\n", fires)
}
